crates/ethrpc/src/http.rs

```diff
@@ -72,19 +72,21 @@ async fn execute_rpc<T: DeserializeOwned>(
 ) -> Result<T, Web3Error> {
     let body = serde_json::to_string(&request)?;
     tracing::trace!(name = %inner.name, %id, %body, "executing request");
-    let mut request = client
+    let mut request_builder = client
         .post(inner.url.clone())
         .header(header::CONTENT_TYPE, "application/json")
         .header("X-RPC-REQUEST-ID", id.to_string())
         .body(body);
-    if let Some(request_id) = observe::request_id::get_task_local_storage() {
-        request = request.header("X-REQUEST-ID", request_id);
+    if let Some(metadata) = observe::request_id::get_task_local_storage() {
+        request_builder = request_builder.header("X-RPC-METADATA", metadata);
     }
```

Comment:
`x-request-id` header is not parsed anymore, which should probably be expected since this function is used only from the buffered transport.